src/classes/window/controllable.vala

```diff
@@ -310,7 +310,9 @@ namespace pdfpc.Window {
 
                 // Punctuation characters are usually generated by keyboards
                 // with the Shift mod pressed; we ignore it in this case.
-                if (((char) key.keyval).ispunct()) {
+                // AZERTY keyboards (and maybe some others) do same for digits.
```

Comment:
I don't have a AZERTY keyboard but does thing change has maybe side effects with "normal" keyboard layouts?

Reply:
I don't have such a hardware keyboard, either (hence, I asked the OP to test this patch). But I used the "French (AZERTY)" as an Input source in the Ubuntu preferences to test, and it works correctly, AFAICT.

I believe the only case where this might have a side effect is a keyboard that has "pure" number keys (i.e., that types, e.g., "1" whether the shift button is pressed or not) AND the user wants to assign a specific action to such a combination. This sounds like a much less probable situation than an AZERTY keyboard (until yesterday, I admit, I hadn't realized that every French-speaking person in the world has to work twice as hard as the rest of us to enter numbers :)).

Reply:
So using a separate num pad could cause some hazzle? This sounds like a minor problem to me.

Reply:
Good question. Unfortunately, I don't have such a device around to test. But I guess it should work like a normal numpad on a full-size keyboard when the "NumLock" is enabled, i.e., reporting (yeah, I should've said "reports" instead of "types" in the previous comment) `KP_1` instead of `1` etc.